extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultBaseProxy.java

```diff
@@ -0,0 +1,86 @@
+package io.quarkus.vault;
+
+import static io.quarkus.vault.VaultProxy.X_VAULT_TOKEN;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import io.quarkus.vault.proxy.dto.VaultCreds;
+import io.quarkus.vault.proxy.dto.VaultHealth;
+import io.quarkus.vault.proxy.dto.VaultInit;
+import io.quarkus.vault.proxy.dto.VaultInitBody;
+import io.quarkus.vault.proxy.dto.VaultKubernetesAuth;
+import io.quarkus.vault.proxy.dto.VaultKubernetesAuthBody;
+import io.quarkus.vault.proxy.dto.VaultKvSecretV1;
+import io.quarkus.vault.proxy.dto.VaultKvSecretV2;
+import io.quarkus.vault.proxy.dto.VaultLeasesBody;
+import io.quarkus.vault.proxy.dto.VaultLeasesLookup;
+import io.quarkus.vault.proxy.dto.VaultLookupSelf;
+import io.quarkus.vault.proxy.dto.VaultRenewLease;
+import io.quarkus.vault.proxy.dto.VaultRenewSelf;
+import io.quarkus.vault.proxy.dto.VaultRenewSelfBody;
+import io.quarkus.vault.proxy.dto.VaultSealStatus;
+import io.quarkus.vault.proxy.dto.VaultUserPassAuth;
+import io.quarkus.vault.proxy.dto.VaultUserPassAuthBody;
+
+public interface VaultBaseProxy {
```

Comment:
Can you please consider having the Kubernetes and UserPass specific extensions instead of having it all in the same interface ? Here one would have only the methods which are relevant after the specific login has occurred

Reply:
if we wanted to split interfaces, I believe it would make more sense to follow vault's api's organization. for instance k8s and userpass login are both part of the auth api: https://www.vaultproject.io/docs/auth/index.html
for instance bettercloud locates all of its login methods in the same class: https://github.com/BetterCloud/vault-java-driver/blob/master/src/main/java/com/bettercloud/vault/api/Auth.java

Reply:
Sure, lets keep it simple for now